kube-core/src/params.rs

```diff
@@ -327,6 +365,13 @@ impl PatchParams {
         self.dry_run = true;
         self
     }
+
+    /// Set the validation directive for `fieldValidation` during server-side apply.
+    #[must_use]
+    pub fn field_validation(mut self, directive: ValidationDirective) -> Self {
+        self.field_validation = Some(directive);
+        self
+    }
```

Comment:
I wonder if it makes sense to have aliased setters for each enum variant (since the enum is small):

- `PatchParams::default().validation_ignore()`
- `PatchParams::default().validation_strict()`
- `PatchParams::default().validation_warn()`

Not 100% sure of this, but it is a bit shorter, and avoids people having to import this new enum into scope. WDYT?

Reply:
I think it's absolutely fine either way, and if dedicated setters is more inline with the rest of the library I'd be happy to adapt that 👍

Reply:
I did just realise that the `field_validation()` method could be more inline with the go library and make usage easier (I'd need to double check), not sure if that's something we care about in kube-rs?

Reply:
yeah, we try to not deviate too much on the basics at least. less concepts for people to learn coming over from go land.
rust ergonomics concerns tends to come first if there's tough choices though.

what do you have in mind?

Reply:
Just meant it's worthwhile me checking if the go lib has conveniences around the field validation and if so aligning with that

Reply:
had a quick look:
- [PatchOptions](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#PatchOptions) (equivalent to our `PatchParams`) has the `FieldValidation` string
- [no useful helpers in apimachinery](https://github.com/kubernetes/apimachinery/search?q=FieldValidation)
- [metav1.FieldValidationStrict](https://github.com/kubernetes/apimachinery/blob/b85d88926e14f20152645d6b697d63dda7929f29/pkg/apis/meta/v1/types.go#L510-L517) and the enum equivalent constants, are used directly

so i think our setup here is pretty good with respect to parity.

(the main lacks in `PatchParams` is probably more utilities around ManagedFields atm - which i believe have some helpers somewhere to allow set merging - but that's completely orthogonal to this pr)